airflow/models.py

```diff
@@ -3678,7 +3757,7 @@ def refresh_from_db(self, session=None):
             DR.dag_id == self.dag_id,
             DR.execution_date == self.execution_date,
             DR.run_id == self.run_id
-        ).one()
+        ).first()
```

Comment:
Oh okay, my bad didn't I catch that when I was reading through the code.
Please go ahead with the revert then

On Wednesday, September 7, 2016, bolkedebruin <notifications@github.com>
wrote:

> In airflow/models.py
> <https://github.com/apache/incubator-airflow/pull/1730#discussion_r77948259>
> :
>
> > @@ -3678,7 +3757,7 @@ def refresh_from_db(self, session=None):
> >              DR.dag_id == self.dag_id,
> >              DR.execution_date == self.execution_date,
> >              DR.run_id == self.run_id
> > -        ).one()
> > +        ).first()
>
> It does. When you self.dag_id, self.execution_date, and self.run_id are
> not present anymore in the db it should throw an exception. This is an
> integrity issue. I think it should be reverted to the old behavior. Airflow
> should throw an exception when the dag_run is not present anymore in the db
> while it still has it in memory.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/incubator-airflow/pull/1730/files/a03fe0478cf7627255782696f083de4deeb8640e#r77948259>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGYMpi14K3ckSwCvTy_Eco-a3TW-8Rk9ks5qn56ugaJpZM4JjjUq>
> .
>


-- 
Thank you,

Norman Mu